### ISSUE TITLE
POM enable the SpringBoot-Data-RestAPI-H2Db-ZeroCode project

### ISSUE THREAD

**Issue by authorjapps** [authorjapps/spring-boot-integration-test#6]
**Objective:**
Create a POM version of this project from Sidhaye's GitLab
https://gitlab.com/Sidhaye/springboot/tree/SpringBoot-RestAPI-ZeroCode-Integration

**TODO things to resolve this ticket:**
+ Request @BeTheCodeWithYou to grant contributor access to us (@authorjapps , @macrocks , @officiallysameer )
+ Add a POM file with exact same dependencies of [Gradle file](https://gitlab.com/Sidhaye/springboot/blob/SpringBoot-RestAPI-ZeroCode-Integration/build.gradle), but use latest version of Zerocode(1.2.x)
+ Add a Suite runner to run all the tests at once
   + Via JUnit Suite class
   + Via Zerocode Package Runner
+ Verify `mvn clean install` works.
   + Tests are happy
   + Reports are generated
+ Share in the README file(specifying Gradle n POM enabled)  and close this.
+ Tweet this link/repo

Basically add a POM file, no code changes needed.

**Comment by BeTheCodeWithYou** [comment]
Hi  Team- @authorjapps , @macrocks , @officiallysameer
I have done required changes to mavenize it. 
Not sure why but gitlab is not able find users with your names. 
![invite_members](https://user-images.githubusercontent.com/23066319/47269494-597bf600-d556-11e8-8a16-a2697afc0cea.JPG)

### **Anyways, below are the change details**

New branch with mavenized project including README.md

**https://gitlab.com/Sidhaye/springboot/tree/API-SpringData-H2DB-ZeroCode-mvn**

**mvn clean install = Build-success**
![build success](https://user-images.githubusercontent.com/23066319/47269468-e1153500-d555-11e8-949d-ffe68659b845.JPG)

**All TEST passed**
![testreport](https://user-images.githubusercontent.com/23066319/47269474-ed00f700-d555-11e8-8444-293efc59d7a4.JPG)

**Java -jar SpringbootRestInMemoryDB-1.0.0-SNAPSHOT.jar**
all up and running

![server-start](https://user-images.githubusercontent.com/23066319/47269481-04d87b00-d556-11e8-8377-18d19a318546.JPG)

**Comment by BeTheCodeWithYou** [comment]
sorry.. missed to add "JUnit Suite class". doing it now.

**Comment by BeTheCodeWithYou** [comment]
Hi Team- @authorjapps , @macrocks , @officiallysameer

Test suite also added.

Cheers have a great time. Awaiting your tweet!  :+1:

**Comment by authorjapps** [comment]
Looks good mate, nice work 👍 . Appreciate your time n effort for this. It's basically we all are trying to make the best practices and best experiences reach all developers/testers, so we need to help each other.

We have a bit of suggestion in terms of organizing the tests like below, just adding a new folder `crudtests` as below and keeping the `Suite` runner outside the `crudtests` folder. 

```
+integrationtest
|
|--crudtests
|	    |
|	    |-- TestGetOperations.java
|	    |-- TestPatchOperation.java
|	    |-- TestPostAndThenGetOperation.java
|	    |-- TestPostOperation.java
|	    
|--IntegrationTestSuite.java
|
```
(We will tweet with this changes done, as this will have almost everything to refer as a sample project to start with for Spring boot developers).

One more request, can you import this Spring project into your GitHub repo(I think GitHub and GitLab projects allows to import each other's repo as they are both `Git` based), as we all have GitHub accounts and it's much easier to collaborate, once you give us access.

It's not super urgent, but when you get chance please 🙏.

Once this is done, we can(collaborate) add add the env switching to reuse the tests for other envs like `CI/DIT/SIT(ci2/dit2/sit2 etc)`. This is super easy with Zerocode(just one more annotation on the suite-runner - _The entire plumbing work then has already taken care in the framework_), but that's not the point. The point is we can make this project an ideal reference project for anyone in the `spring-boot` community(both developer as well as testers to clone and do their job easily in just matter of minutes. _(we will raise a separate ticket to this bit, and close the current one)_

Does it sound ok ?

**Comment by BeTheCodeWithYou** [comment]
Hi @authorjapps 

All sorted! 

1) updated pom.xml to use 1.2.8
2) Added reuse of request body JSON.
3) update property file to use web instead restful.
4) added cit and sit property file.
5) added @EnvProperty("_${env}")  to IntegrationTestSuite
6) Tested evn with below approach
7) All code pushed and repo is import to github
https://github.com/BeTheCodeWithYou/springboot/tree/API-SpringData-H2DB-ZeroCode-mvn

**A** = **mvn clean install**    ( performs all test against application_host.properties and genrates application JAR )

**B** = running 2 instances of application JAR on differnt ports

**java -jar -Dserver.port=9090 SpringbootRestInMemoryDB-1.0.0-SNAPSHOT.jar**

**java -jar -Dserver.port=7070 SpringbootRestInMemoryDB-1.0.0-SNAPSHOT.jar**  


**C** = Running IntegrationTestSuite against test environment profile

**mvn -Denv=cit -Dtest=IntegrationTestSuite test**    
( runs test against application_host_cit.properties. Can see logs on the JAR running on 9090 port. SUCCESS.)

**mvn -Denv=sit -Dtest=IntegrationTestSuite test**
( runs test against application_host_sit.properties. Can see logs on the JAR running on 7070 port. SUCCESS.)

**mvn -Dtest=IntegrationTestSuite test**
( runs test against application_host.properties which is on 8080 setup. SUCCESS.)

![zerocode_profile](https://user-images.githubusercontent.com/23066319/47322119-c2df3000-d64e-11e8-8f6a-27f8276ad980.JPG)


have a great day champs! Cheers.

**Comment by BeTheCodeWithYou** [comment]
@authorjapps   can we close this issue, as the necessary code is in place inside "spring boot integration repo". Cheers!
